src/shared/property.js

```diff
@@ -250,3 +250,28 @@ export const propertySum = (left, right, transform) => {
   right.subscribe((r) => inner.set(transform(left.value, r)));
   return inner.readOnly;
 };
+
+/**
+ * Creates a "sum-type" property.
+ * Takes 3 Properties {L,M,R} and a function (l,m,r)=>T
+ * When L, M, or R changes calls the function
+ * and updates the returned property.
+ *
+ *
+ * @template T
+ * @template L
+ * @template M
+ * @template R
+ * @param {IBindable<L>} left - Left Hand Property
+ * @param {IBindable<R>} right - Middle Hand Property
+ * @param {IBindable<R>} right - Right Hand Property
+ * @param {(arg0: L, arg1: M, arg2: R)=>T} transform - Called with the Property Value, must return the transformed value
+ * @returns {ReadOnlyProperty<T>} -
+ */
+export const propertySumTrio = (left, middle, right, transform) => {
```

Comment:
A: First of really appreciate that you looked into that system :D 
However i dont think that scales. 
Consider cherry picking https://github.com/mozilla-extensions/mozilla-vpn-extension/pull/144 - i think we can just rewrite property sum to take N amount of properties :)

Reply:
Thanks! I knew it was only a quick band aid fix... Will cherry pick that improved one after it is merged in.